src/server/game/Scripting/ScriptMgr.h

```diff
@@ -1013,6 +1016,7 @@ class TC_GAME_API ScriptMgr
         void OnPlayerUpdateZone(Player* player, uint32 newZone, uint32 newArea);
         void OnQuestStatusChange(Player* player, uint32 questId);
         void OnPlayerRepop(Player* player);
+        void OnPlayerReachedQuestObjectiveCount(Player* player, uint32 questId, uint32 creatureOrGoEntry, uint8 count);
```

Comment:
I would rename this to `OnQuestObjectiveProgress`, and change arguments
* `uint32 questId` to `Quest const* quest`
* `uint32 creatureOrGoEntry` to `uint32 objectiveIndex` (see the other comment)
* `uint8 count` to `uint16 progress` (rename + change type, updatefields store uint16s)